From 4f35388cda2ed4f314000788136616dfa7f7a54c Mon Sep 17 00:00:00 2001
From: Alec Ari <neotheuser@ymail.com>
Date: Wed, 16 Oct 2019 16:48:34 -0500
Subject: Fix build with new RTAI on Linux 4.14.148

LinuxCNC needs the following commits:

(Only included in LinuxCNC master and 2.8 git branches)
469d3cb9b767a48d149b4ab2a02fe49fb07eae26
pci_8255: Change function names to avoid clash with v4.x kernel macros.

(Included in LinuxCNC master, 2.8 and 2.7 git branches)
0dbbfb3ae106feea27dab4753ae40c21b6ea06d3
RTAI_RTAPI: RTAPI does not compile with kernel 4.14 without this change.

Signed-off-by: Alec Ari <neotheuser@ymail.com>
---
 src/configure.ac | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/src/configure.ac b/src/configure.ac
index 3d0f621a6..7005fcf63 100644
--- a/src/configure.ac
+++ b/src/configure.ac
@@ -371,6 +371,7 @@ case $RTS in
     RTPREFIX=rtai
     RTAI_VERSION=$($RTS --version)
     RTAI=$(echo $RTAI_VERSION | cut -f 1 -d .)
+    RTAI_LINUXVER=`$RTS --full-linux-version`
     RTDIR=`$RTS --prefix`
     RTDIR=$(cd $RTDIR ; pwd -P )
     RTFLAGS=`$RTS --module-cflags`
@@ -593,15 +594,7 @@ install with "sudo apt-get install libusb-1.0-0-dev" or disable with
 
 if test $RTS '!=' uspace; then
 AC_MSG_CHECKING([for kernel version string])
-
-if test -e $KERNELDIR/include/linux/utsrelease.h; then
-    VERSION_FILE=$KERNELDIR/include/linux/utsrelease.h
-elif test -e $KERNELDIR/include/generated/utsrelease.h; then
-    VERSION_FILE=$KERNELDIR/include/generated/utsrelease.h
-else
-    VERSION_FILE=$KERNELDIR/include/linux/version.h
-fi
-KERNEL_VERS=`$CC -E -dM ${VERSION_FILE} | grep UTS | cut -d '"' -f 2`
+KERNEL_VERS="${RTAI_LINUXVER}"
 
 if test -z "$KERNEL_VERS"; then 
     AC_MSG_ERROR(Kernel version string not found)
-- 
2.22.0

